src/coreclr/gc/gcpriv.h

```diff
@@ -1154,6 +1155,41 @@ enum interesting_data_point
     max_idp_count
 };
 
+#ifdef USE_REGIONS
+enum free_region_kind
+{
+    basic_free_region,
+    large_free_region,
+    huge_free_region,
+    count_free_region_kinds,
+};
+
+class region_free_list
+{
+    size_t  num_free_regions;
+    size_t  size_free_regions;
+    size_t  size_committed_in_free_regions;
+    size_t  num_free_regions_added;
+    size_t  num_free_regions_removed;
+    heap_segment* head_free_region;
+    heap_segment* tail_free_region;
+
+    static size_t get_region_committed_size (heap_segment* region);
```

Comment:
there's already a `get_region_size` global func, this could be a global func as well? it does what `get_region_size` does just for committed.

Reply:
`size_committed_in_free_regions` doesn't seem to be used? I see it gets updated when we do add/remove regions from these region free lists but it's not taken into consideration when we actually decide how much to decommit. I can imagine a scenario where we have some huge regions but not much is committed on them; and we'd decommit a bunch of large regions because they happen to be at higher address range.

Reply:
Ok, I made get_region_committed_size into a global.

I carried over size_committed_in_free_regions from earlier code. I don't yet have a plan how to make use of it - this needs more thought.